### PR TITLE
Convert cli/test_fact.py to pytest

### PR DESCRIPTION
**Pull request by mshriver**
```
setup@localhost:~/repos/robottelo (pytest-test-fact-13982$%<>) % pytest -vx tests/foreman/cli/test_fact.py 
======================================================================================== test session starts =========================================================================================
platform linux -- Python 3.8.8, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /home/setup/repos/robottelo/.robottelo/bin/python3.8
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, ibutsu-1.14.1, xdist-2.2.1, mock-3.5.1, reportportal-5.0.8, cov-2.11.1, services-2.2.1
collected 6 items                                                                                                                                                                                    

tests/foreman/cli/test_fact.py::test_positive_list_by_name[uptime] PASSED                                                                                                                      [ 16%]
tests/foreman/cli/test_fact.py::test_positive_list_by_name[uptime_days] PASSED                                                                                                                 [ 33%]
tests/foreman/cli/test_fact.py::test_positive_list_by_name[uptime_seconds] PASSED                                                                                                              [ 50%]
tests/foreman/cli/test_fact.py::test_positive_list_by_name[memoryfree] PASSED                                                                                                                  [ 66%]
tests/foreman/cli/test_fact.py::test_positive_list_by_name[ipaddress] PASSED                                                                                                                   [ 83%]
tests/foreman/cli/test_fact.py::test_negative_list_by_name PASSED                                                                                                                              [100%]

========================================================================================= 6 passed in 51.87s =========================================================================================
```

### CODE REVIEW

**Comment by mshriver** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_fact.py

File: tests/foreman/cli/test_fact.py [hunk 1]
```diff
@@ -12,49 +12,41 @@
...
-        :CaseImportance: Critical
-        """
-        for fact in ('uptime', 'uptime_days', 'uptime_seconds', 'memoryfree', 'ipaddress'):
-            with self.subTest(fact):
-                args = {'search': f"fact={fact}"}
-                facts = Fact().list(args)
-                self.assertEqual(facts[0]['fact'], fact)
+    :parametrized: yes
+    """
+    facts = Fact().list(options={'search': f"fact={fact}"})
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_fact.py:41]: I think you have an idea what I'm going to say 🙃

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
Small change

**Review by JacobCallahan — APPROVED** [review]

File: tests/foreman/cli/test_fact.py [hunk 2]
```diff
@@ -12,49 +12,41 @@
...
 import pytest
 from fauxfactory import gen_string
 
 from robottelo.cli.fact import Fact
-from robottelo.test import CLITestCase
 
 
-class FactTestCase(CLITestCase):
-    """Fact related tests."""
+pytestmark = [pytest.mark.tier1]
```
Comment (jameerpathan111) [inline tests/foreman/cli/test_fact.py:25]: @mshriver can you tell me how this work? I mean why it's here and why we don't need to mark test like the way we used to?

Comment (mshriver) [reply tests/foreman/cli/test_fact.py:25]: https://docs.pytest.org/en/stable/reference.html#globalvar-pytestmark

Sure!  pytest will read the `pytestmark` variable from a test module and apply the listed mark decorators to all of the classes/functions within the module.

Just like the testimony tokens, the marks don't have to be repeated on every test.

**Review by jameerpathan111 — APPROVED** [review]
Ack

**Pull request merged**